prime-router/src/main/kotlin/SubmissionReceiver.kt

```diff
@@ -256,6 +256,7 @@ class ELRReceiver : SubmissionReceiver {
         payloadName: String?,
         metadata: Metadata?
     ) {
+        check(sender is FullELRSender)
```

Comment:
Basically this should only be used by `FullELRSender`s, but we didn't previously need to smart cast it to a `FullELRSender` because everything we needed is in the top level `Sender` definition. With the usage of `schemaName` at line 346, we do need to cast `sender`, and figured rather than do that locally there, we might as well add this check at the entry point to the function and be able to use `sender` as a `FullELRSender` throughout.

Reply:
See my comment regarding `schemaName`

Reply:
I think that's a good call to pull `schemaName` into the top level `Sender`. I'm inclined to keep this check here either way, so let me know if you think otherwise.